CMake/SetTargetSourceFolder.cmake

```diff
@@ -0,0 +1,19 @@
+function(set_target_source_folder target)
```

Comment:
All of this is good, but the it was unclear to me from the name alone what the function does. What about something like `dolphin_add_target_sources_to_ide_folders`?